### PR TITLE
Switch embed fields to interactive buttons

### LINKED ISSUE #850: [FEAT] - Donate Command
### What feature would you like to see?

The donate command should be updated to use link buttons!

Within the [handleDonate.ts file](https://github.com/BeccaLyria/discord-bot/blob/main/src/modules/commands/subcommands/becca/handleDonate.ts), create three new `MessageButton()` components using the `link` style, one for each of the fields on the embed.

The buttons need to be added to a `MessageActionRow()`, which gets passed to the `interaction.editReply()` call as the `components` property (in an array).

The fields can then be removed from the embed.

### Additional information

[Example](https://github.com/RosaliaNightsong/discord-bot/blob/main/src/modules/subcommands/rosalia/handleDonate.ts)

### PR DESCRIPTION
**Pull request by DeltaXWizard**
# Pull Request

## Description

This implements interactive buttons which replace the current embed fields.

## Related Issue

Closes #850

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [X] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/modules/commands/subcommands/becca/handleDonate.ts [hunk 1]
```diff
@@ -17,17 +17,30 @@ export const handleDonate: CommandHandler = async (Becca, interaction) => {
...
-    );
-    sponsorEmbed.addField("Patreon", "https://patreon.com/nhcarrigan");
-    sponsorEmbed.addField("PayPal", "https://paypal.me/nhcarrigan");
     sponsorEmbed.setFooter(
       "Join our Discord to get perks when you become a monthly sponsor!"
     );
 
-    await interaction.editReply({ embeds: [sponsorEmbed] });
+    const githubButton = new MessageButton()
+      .setLabel("GitHub Sponsors")
```
Comment (nhcarrigan) [inline src/modules/commands/subcommands/becca/handleDonate.ts:25]:
```suggestion
      .setLabel("Donate on GitHub!")
```

File: src/modules/commands/subcommands/becca/handleDonate.ts [hunk 2]
```diff
@@ -17,17 +17,30 @@ export const handleDonate: CommandHandler = async (Becca, interaction) => {
...
+      .setStyle("LINK")
+      .setURL("https://github.com/sponsors/nhcarrigan");
+    const patreonButton = new MessageButton()
+      .setLabel("Patreon")
```
Comment (nhcarrigan) [inline src/modules/commands/subcommands/becca/handleDonate.ts:29]:
```suggestion
      .setLabel("Donate on Patreon!")
```

File: src/modules/commands/subcommands/becca/handleDonate.ts [hunk 3]
```diff
@@ -17,17 +17,30 @@ export const handleDonate: CommandHandler = async (Becca, interaction) => {
...
+      .setStyle("LINK")
+      .setURL("https://www.patreon.com/nhcarrigan");
+    const paypalButton = new MessageButton()
+      .setLabel("Paypal")
```
Comment (nhcarrigan) [inline src/modules/commands/subcommands/becca/handleDonate.ts:33]:
```suggestion
      .setLabel("Donate on Paypal!")
```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
Quick verbiage tweak to make these CTAs.

**Comment by DeltaXWizard** [comment]
Fixed.

**Review by nhcarrigan — APPROVED** [review]
Awesome. Looks good to me.

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**